core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -176,6 +178,15 @@ public int getPort() {
     }
   }
 
+  @Override
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
+  public List<ServerServiceDefinition> getServices() {
+    List<ServerServiceDefinition> services =
```

Comment:
This needs to be split into static and dynamic parts.   If the fallbackRegistry is empty, there is no need to do an extra allocation.  If the mutable handler registry is not empty, you can pre allocate the correct size arraylist ahead of time.